packages/client/src/components/settings-dialog.vue

```diff
@@ -101,14 +96,36 @@ const { dialogRef, onDialogCancel, onDialogOK, onDialogHide } =
 
 const { t } = useI18n();
 
-const newSettings = reactive<SettingsUpdateInput>({
+const newSettings = reactive<RetailLocationSettingsFragment>({
   maxBookingDays: props.maxBookingDays,
   payOffEnabled: props.payOffEnabled,
   warehouseMaxBlockSize: props.warehouseMaxBlockSize,
+  buyRate: props.buyRate,
+  sellRate: props.sellRate,
 });
 
-const purchaseRate = props.buyRate;
-const saleRate = props.sellRate;
+function validateSettings() {
+  const settingsUpdate = {
+    type: "save" as const,
+    settings: newSettings,
+  };
+
+  if (
+    newSettings.buyRate !== props.buyRate ||
+    newSettings.sellRate !== props.sellRate
+  ) {
+    Dialog.create({
+      title: t("general.settings.updateRatesConfirmTitle"),
+      message: t("general.settings.updateRatesConfirmMessage"),
+      ok: t("actions.continue"),
```

Comment:
I would rather align the button name to title of the dialog which is update and not continue. So let's use `UPDATE` and not `CONTINUE` as verb here.